core/domain/stats_jobs_one_off.py

```diff
@@ -307,11 +307,18 @@ def reduce(exp_id, stringified_values):
 
             # Compute total_hit_count and first_hit_count for the states.
             for state_name in state_hit_counts_for_this_version:
-                state_stats_mapping[state_name].total_hit_count_v1 += (
-                    state_hit_counts_for_this_version[state_name][
-                        'total_hit_count'])
-                state_stats_mapping[state_name].first_hit_count_v1 += (
-                    len(state_session_ids_by_version[version][state_name]))
+                try:
+                    state_stats_mapping[state_name].total_hit_count_v1 += (
+                        state_hit_counts_for_this_version[state_name][
+                            'total_hit_count'])
+                    state_stats_mapping[state_name].first_hit_count_v1 += (
+                        len(state_session_ids_by_version[version][state_name]))
+                except KeyError:
+                    yield (
+                        'ERROR: State not in stats mapping exp_id %s, version '
+                        '%s, State %s, state_stats_mapping [%s]' % (
+                            exp_id, version, state_name,
```

Comment:
Take a look at some of the other jobs -- I think you'll need to encode stuff like state_name with utf-8 because sometimes it is unicode. Ditto for some of the other stuff; try to test locally with unicode state names (e.g. with Chinese/Hindi chars).

Reply:
Done.